### PR TITLE
Add general philosophy and improve formatting

### PR DESCRIPTION
**Pull request by nemild**

### CODE REVIEW

File: smart-contracts.md [hunk 1]
```diff
@@ -1,6 +1,57 @@
...
+
+## Security Notifications
+
+This is a list of resources that will often highlight discovered exploits in Ethereum or Solidity:
+
+- [Ethereum Blog](https://blog.ethereum.org/): The official Ethereum blog
+- [Hacking Distributed](http://hackingdistributed.com/): Professor Sirer's blog with regular posts on cryptocurrencies and security
+- [Vessenes.com](http://vessenes.com/): Peter Vessenes blog
+- [Network Stats](https://ethstats.net/)
+
```
Comment (simondlr) [inline smart-contracts.md]: I'd suggest adding in Reddit and the Gitter channels of Ethereum (Solidity/Research/Go-ethereum/cpp-ethereum etc).

File: smart-contracts.md [hunk 2]
```diff
@@ -1,6 +1,57 @@
...
+It's highly recommended that you *regularly* read all these sources, as exploits they note may impact your contracts.
+
+Additionally, this is a list of community members who may write about security:
+
+- **Vitalik Buterin**: [Twitter](https://twitter.com/vitalikbuterin), [Github](https://github.com/vbuterin), [Reddit](https://www.reddit.com/user/vbuterin), [Ethereum Blog](https://blog.ethereum.org/author/vitalik-buterin/)
+- **Dr. Christian Reitwiessner**: [Twitter](https://twitter.com/ethchris), [Github](https://github.com/chriseth), [Ethereum Blog](https://blog.ethereum.org/author/christian_r/)
+- **Gavin Wood**: [Twitter](https://twitter.com/gavofyork), [Blog](http://gavwood.com/), [Github](https://github.com/gavofyork)
+- **Dr. Emin Gun Sirer**: [Twitter](https://twitter.com/el33th4xor)
+- **Vladimir Zamfir**: [Twitter](https://twitter.com/vladzamfir), [Github](https://github.com/vladzamfir), [Ethereum Blog](https://blog.ethereum.org/author/vlad/)
```
Comment (simondlr) [inline smart-contracts.md]: Is his full Vladimir? I think everyone just calls him Vlad. :)

File: smart-contracts.md [hunk 3]
```diff
@@ -1,6 +1,57 @@
...
+- [Vessenes.com](http://vessenes.com/): Peter Vessenes blog
+- [Network Stats](https://ethstats.net/)
+
+It's highly recommended that you *regularly* read all these sources, as exploits they note may impact your contracts.
+
+Additionally, this is a list of community members who may write about security:
+
+- **Vitalik Buterin**: [Twitter](https://twitter.com/vitalikbuterin), [Github](https://github.com/vbuterin), [Reddit](https://www.reddit.com/user/vbuterin), [Ethereum Blog](https://blog.ethereum.org/author/vitalik-buterin/)
+- **Dr. Christian Reitwiessner**: [Twitter](https://twitter.com/ethchris), [Github](https://github.com/chriseth), [Ethereum Blog](https://blog.ethereum.org/author/christian_r/)
+- **Gavin Wood**: [Twitter](https://twitter.com/gavofyork), [Blog](http://gavwood.com/), [Github](https://github.com/gavofyork)
```
Comment (simondlr) [inline smart-contracts.md]: Gavin is also a Dr.

File: smart-contracts.md [hunk 4]
```diff
@@ -155,129 +232,105 @@ A way around this could be to use block numbers and estimate time that has passe
...
 
-Essentially, the guarantees are provided by a richer type system. Ensuring that a program/function is properly typed proves that this formula has a certain structure.
 
-The type system of procedural languages like Solidity are less fine-grained than those of functional languages.
+### Automatic deprecation
 
-With Solidity, you can ensure that a function that returns a uint256 will not return an address, for example. (notice that there is still no guarantee that the function wouldn’t throw, whereas in functional languages this is usually guaranteed).
 
-With a more sophisticated type system, rich enough to express behavior such as ‘fungible resources’, (in short, a resource that cannot be copied, see resource on linear types below), one can prove more advanced properties of functions.
+### Versioning
```
Comment (simondlr) [inline smart-contracts.md]: Is the idea that these are still to be expanded?

**Comment by simondlr** [comment]
@nemild this is excellent restructuring. Thank you so much. I commented on some parts of it!

File: smart-contracts.md (hunk 4 above)
Comment (nemild) [inline smart-contracts.md]: Yes, that's correct - this is just to get us to start a discussion before writing the exact text

**Comment by nemild** [comment]
Thanks for the comments, incorporated all of them.

**Pull request merged**